app/controllers/coronavirus_form/able_to_leave_controller.rb

```diff
@@ -20,12 +20,24 @@ def submit
       render controller_path
     else
       update_session_store
+      write_responses
       redirect_to results_path
     end
   end
 
 private
 
+  # We're using this method to reduce the precision of timekeeping so that
+  # responses cannot be correlated with analytics data
+  def time_hour_floor
+    time = Time.current
+    time - time.sec - time.min % 60 * 60
```

Comment:
Have you seen this: https://api.rubyonrails.org/classes/Time.html#method-i-beginning_of_hour ?

Reply:
Gives you this:

```
[1] pry(main)> time = Time.current
=> Thu, 09 Apr 2020 15:51:03 UTC +00:00
[2] pry(main)> time.beginning_of_hour
=> Thu, 09 Apr 2020 15:00:00 UTC +00:00
[3] pry(main)> 
```

Reply:
well that does exactly what we wanted, nice spot!

Reply:
done!

Reply:
👨‍🍳💋  - nice! We looked at Ruby's time class, didn't realise Rails had monkey patched it for us!

Reply:
Rule of thumb with Rails is, if you think there should be a nice way to do something, someone has probably already written it.